### PR TITLE
[UNR-5632] Add codegen tool

### PR DESCRIPTION
**Pull request by ImprobableNic**
#### Description
Add a little codegen tool that helps quickly get the C++ counterpart of a user-defined component.
This only supports a couple of base types, and is manually ran as to be as unobtrusive as possible.

### CODE REVIEW

**Comment by simonsarginson** [comment]
Is there a reason this is written in cxx ? If it were written in say Python it would be pretty trivial to run outside of the context of the editor? UE4 has support for invoking python scripts from editor too I am pretty sure.

**Comment by ImprobableNic** [comment]
Well, I would have been slower at writing this in Python compared to cxx, but that's just me.

**Comment by simonsarginson** [comment]
Fair enough

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKCodegenTool.cpp [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+			{
+				FString ErrorMessage =
+					FString::Printf(TEXT("Type %s part of component %s cannot be handled."), *Field.Type, *Component.Id.Name);
+				FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(ErrorMessage));
+				bAllTypesHandled = false;
+				break;
+			}
+
+			FString FieldName = Field.Name;
+			FieldName[0] = toupper(FieldName[0]);
```
Comment (simonsarginson) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKCodegenTool.cpp:277]:
```suggestion
			FieldName[0] = FieldName[0].ToUpper();
```

Comment (ImprobableNic) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKCodegenTool.cpp:277]: I just want the first character to be converted, ToUpperInline seems to operate over the entire string?

**Pull request closed without merging**